### ISSUE TITLE
Error: Access Denied when attempting to download audiobooks on motorola one 5g ace

### ISSUE THREAD

**Issue by cassieesposito** [advplyr/audiobookshelf-app#115]
### Steps to reproduce

1. Open audiobookshelf
2. Go to page for particular audiobook
3. Click download button
4. Select a folder to Download in.
5. Touch "USE THIS FOLDER"
6. Touch "ALLOW" on dialog "Allow AudioBookshelf to access files in [folder name]"

### Expected behaviour
- Access granted, audiobook starts downloading

### Actual behaviour
- Red notice at the bottom of the screen that says "Error: Access Denied"

### Troubleshooting steps already attempted
1. Open system settings and grant "Storage" permissions to Audiobookshelf app manually. No change to behavior.

### Environment data

Audiobookshelf Server Version: 1.7.1, Debian package
Audiobookshelf App Version: 0.9.40-beta
Android version: Android 10
Device model: Motorola One 5g Ace
Stock or customized system: Stock

**Comment by advplyr** [comment]
Is this happening for every directory you try?

**Comment by cassieesposito** [comment]
Yes. I have tried multiple directories. I have tried pre-existing directories. I tried newly created directories. I tried external SD card and internal storage. I tried a subfolder of "Downloads" because I was fairly certain that would be allowable for userland apps. All produced the same results.

**Comment by Catsrules** [comment]
I ran into the exact same problem as you on my Galaxy S9. 
I was able to fix it by clearing the data/cache of the app and starting over. That seemed to do the trick for me. 

After I cleared the cache and re-setup the server I first tried the download folder and that worked. I then tried an folder on my SD card but that didn't work I got a permission error. However I had previously gotten a message saying give it permissions on the root of the SD card first. I first select the root of the SD card and that work. After that I could select my Audiobook folder I have on the SD card.

**Comment by JerboaGobi** [comment]
> 

On a Note9 having the same issue. When I first set up the app I attempted to set the download location to a folder on the sdcard. Clearing the data/cache of the app and starting over also did the trick.

- clear app/data cache
- login
- Open Menu > Select Downloads
- Select File > Change Folder
- Navigate to SD Card > Grant access to "SD Card"
- Select Change Folder
- Navigate to SD Card > Subdirectory > Grant Access to "Subdirectory"

Content downloads without issue.

**Comment by bobek** [comment]
I am facing the same issue on 

Audiobookshelf Server Version: 2.0.3, docker
Audiobookshelf App Version: 0.9.42-beta
Android version: Android 10
Device model: OnePlus 5
Stock or customized system: OxygenOS 10.0.1

Despite having grated storage access as well as tapping "allow" after selecting folder, app will show "Error: Access Denied" flash message.

One would almost say, that the Android 10 is a culprit here. I have tried to capture some logs from the device, but I cannot see anything really useful

```
04-26 07:57:28.148 25016 25016 E AbsFileSystem: STORAGE ACCESS DENIED
04-26 07:57:28.561 23536 23536 I Surface : opservice is null false
04-26 07:57:28.563   829  3028 E SurfaceFlinger: Permission Denial: SurfaceFlinger did not recognize request code: 20001
04-26 07:57:28.563  1603  4621 E OnePlusService: Exception notifySurfaceFlingerUpdate
04-26 07:57:28.563  1603  4621 W System.err: java.lang.SecurityException
04-26 07:57:28.563  1603  4621 W System.err: 	at android.os.BinderProxy.transactNative(Native Method)
04-26 07:57:28.563  1603  4621 W System.err: 	at android.os.BinderProxy.transact(BinderProxy.java:557)
04-26 07:57:28.563  1603  4621 W System.err: 	at com.oneplus.server.OnePlusService.notifySurfaceFlingerUpdate(Unknown Source:21)
04-26 07:57:28.563  1603  4621 W System.err: 	at com.oneplus.os.IOnePlusService$Stub.onTransact(IOnePlusService.java:356)
04-26 07:57:28.563  1603  4621 W System.err: 	at android.os.Binder.execTransactInternal(Binder.java:1032)
04-26 07:57:28.564  1603  4621 W System.err: 	at android.os.Binder.execTransact(Binder.java:1005)
```

**Comment by advplyr** [comment]
Please try again with the app version v0.9.43-beta

**Comment by bobek** [comment]
> Please try again with the app version v0.9.43-beta

Unfortunately no change for me. Still getting the access denied error. Is there anything I can share to simplify debugging?

**Comment by advplyr** [comment]
When you first launched the app did you accept the permission request?

I'm not familiar with OxygenOS. Can you go through the settings and try setting permissions for the app?

**Comment by bobek** [comment]
> When you first launched the app did you accept the permission request?

Sure thing. I think that I've done what I should have as a user. Just to be sure, I have
- uninstalled the app, and installed it again from the app store (google play)
- app asks for the storage permissions after launch
  ![Screenshot_20220501-202003](https://user-images.githubusercontent.com/26891/166159235-8fbe9f14-d91d-4338-a2ef-727ed3984784.jpg)
- it is the correct version, so adding the local media path (it doesn't matter if app is connected to the server or not, behavior seems the same)
  ![Screenshot_20220501-202011](https://user-images.githubusercontent.com/26891/166159240-2639fc5b-d253-4042-a527-7bc612b7ec69.jpg)
- Selected folder (`Downloads` in this case) and confirmed. Additional popup shows
  ![Screenshot_20220501-202023](https://user-images.githubusercontent.com/26891/166159241-5405fc37-b2c2-4607-aeab-911931079a23.jpg)
- Tapped `Allow` (`Cancel` closes just the popup and returns to folder selection). Folder is not added and the error flash message is shown.
  ![Screenshot_20220501-202028](https://user-images.githubusercontent.com/26891/166159244-d7851edf-c0ea-4717-921b-b9889e855be9.jpg)
- Just to be sure, checking permissions via the `App info`
  ![Screenshot_20220501-202048](https://user-images.githubusercontent.com/26891/166159245-f5010695-d525-4f58-bce2-d02aa13cd62e.jpg)

**Comment by advplyr** [comment]
Thanks for providing details, that is helpful. I know there is some weirdness around the Download folder specifically. Have you tried other folders that are already on your phone?
There is also a solution posted on discord relating to giving access permission to every system folder below, but I'm not sure how that works: https://discord.com/channels/942908292873723984/942914154254176257/970030898995163226

**Comment by bobek** [comment]
> Thanks for providing details, that is helpful. I know there is some weirdness around the Download folder specifically. Have you tried other folders that are already on your phone? 

Yes, I've tried different folder(s) as well. Behavior is the same, so I have not reported it, but yeah, something else than `Downloads` would be probably less confusing.

> There is also a solution posted on discord relating to giving access permission to every system folder below, but I'm not sure how that works: https://discord.com/channels/942908292873723984/942914154254176257/970030898995163226

That works :+1: . So it is resolved for me, maybe you may consider adding the hint directly into the app, so people try that first before reporting issues.

For the benefit of others, the solution is to
- give permission to `/0/` first
- then to give permission to folder of your choice, e.g. `audiobooks/`
- basically you need to give permissions to every directory along the path to your target one

**Comment by wtanksleyjr** [comment]
Confirm that it works to give permissions to every folder in the path.

**Comment by cassieesposito** [comment]
Resolved in 0.9.46 beta
